staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -244,8 +255,19 @@ func (a customResourceStrategy) ValidateUpdate(ctx context.Context, obj, old run
 }
 
 // WarningsOnUpdate returns warnings for the given update.
-func (customResourceStrategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
-	return nil
+func (a customResourceStrategy) WarningsOnUpdate(ctx context.Context, obj, old runtime.Object) []string {
+	var allWarnings []string
+	fldPath := field.NewPath("metadata")
+	newObjAccessor, err := meta.Accessor(obj)
+	if err != nil {
+		allWarnings = append(allWarnings, field.Invalid(fldPath, nil, err.Error()).Error())
+	}
+	for _, finalizer := range newObjAccessor.GetFinalizers() {
```

Comment:
on update, I think we should only validate _new_ finalizer names added, otherwise every update of an object already containing an unqualified finalizer name will emit a warning to the user doing the update, even if they had nothing to do with the finalizer